editor_dll/CMakeLists.txt

```diff
@@ -0,0 +1,20 @@
+cmake_minimum_required(VERSION 3.6)
+project(editor_dll)
+
+# Include common plugin CMake scripts and set the type of plugin to be to and EDITOR_PLUGIN
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${REPOSITORY_DIR}/cmake")
+include(CMakePlugin)
+include(GenerateExportHeader)
+
+# Scan and add project source files
+find_source_files(ALL_SOURCE_FILES)
+
+# Setup DLL dynamic library
+set(BUILD_SHARED_LIBS TRUE)
+add_library(${PROJECT_NAME} SHARED ${ALL_SOURCE_FILES})
+generate_export_header(${PROJECT_NAME})
```

Comment:
`generate_export_header` will save you some time by generate all the DLL export stuff. e.g.:
```

#ifndef EDITOR_DLL_EXPORT_H
#define EDITOR_DLL_EXPORT_H

#ifdef EDITOR_DLL_STATIC_DEFINE
#  define EDITOR_DLL_EXPORT
#  define EDITOR_DLL_NO_EXPORT
#else
#  ifndef EDITOR_DLL_EXPORT
#    ifdef editor_dll_EXPORTS
        /* We are building this library */
#      define EDITOR_DLL_EXPORT __declspec(dllexport)
#    else
        /* We are using this library */
#      define EDITOR_DLL_EXPORT __declspec(dllimport)
#    endif
#  endif

#  ifndef EDITOR_DLL_NO_EXPORT
#    define EDITOR_DLL_NO_EXPORT 
#  endif
#endif

#ifndef EDITOR_DLL_DEPRECATED
#  define EDITOR_DLL_DEPRECATED __declspec(deprecated)
#endif

#ifndef EDITOR_DLL_DEPRECATED_EXPORT
#  define EDITOR_DLL_DEPRECATED_EXPORT EDITOR_DLL_EXPORT EDITOR_DLL_DEPRECATED
#endif

#ifndef EDITOR_DLL_DEPRECATED_NO_EXPORT
#  define EDITOR_DLL_DEPRECATED_NO_EXPORT EDITOR_DLL_NO_EXPORT EDITOR_DLL_DEPRECATED
#endif

#if 0 /* DEFINE_NO_DEPRECATED */
#  ifndef EDITOR_DLL_NO_DEPRECATED
#    define EDITOR_DLL_NO_DEPRECATED
#  endif
#endif

#endif
```